Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,489 @@
+constructor argument.
+
+..  _extbase_fileupload_attribute-validation:
+
+Validation:
+~~~~~~~~~~~
+
+File upload validation is defined in an array of validators in the
+:php-short:`\TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration` object. The validator
+:php:`\TYPO3\CMS\Extbase\Validation\Validator\FileNameValidator`,
+which ensures that no executable PHP files can
+be uploaded, is added by default if the file upload configuration object
+is created using the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute.
+
+In addition, Extbase includes the following validators to validate an
+:php-short:`\TYPO3\CMS\Core\Http\UploadedFile` object:
+
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\FileSizeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\ImageDimensionsValidator`
+
+Those validators can either be configured with the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute or added
+manually to the configuration object
+with the :php:`addValidator` method.
+
+..  _extbase_fileupload_attribute-required:
+
+Required:
+~~~~~~~~~
+
+Defines whether a file must be uploaded. If it is set to `true`, the
+:php:`minFiles` configuration is set to `1`.
+
+..  _extbase_fileupload_attribute-minimum-files:
+
+Minimum files:
+~~~~~~~~~~~~~~
+
+Defines the minimum amount of files to be uploaded.
+
+..  _extbase_fileupload_attribute-maximum-files:
+
+Maximum files:
+~~~~~~~~~~~~~~
+
+Defines the maximum amount of files to be uploaded.
+
+..  _extbase_fileupload_attribute-upload-folder:
+
+Upload folder:
+~~~~~~~~~~~~~~
+
+Defines the upload path for the file upload. This configuration expects a
+storage identifier (e.g. :php:`1:/user_upload/folder/`). If the given target
+folder in the storage does not exist, it is created automatically.
+
+..  _extbase_fileupload_attribute-upload-folder-creation:
+
+Upload folder creation, when missing:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default creation of a missing storage folder can be disabled via the
+configuration attribute :php:`createUploadFolderIfNotExist`
+(:php:`bool`, default :php:`true`).
+
+..  _extbase_fileupload_attribute-random-suffix:
+
+Add random suffix:
+~~~~~~~~~~~~~~~~~~
+
+When enabled, the filename of an uploaded and persisted file will contain a
+random 16 char suffix. As an example, an uploaded file named
+:php:`job-application.pdf` will be persisted as
+:php:`job-application-<random-hash>.pdf` in the upload folder.
+
+The default value for this configuration is :php:`true` and it is recommended
+to keep this configuration active.
+
+This configuration only has an effect when uploaded files are persisted.
+
+..  _extbase_fileupload_attribute-duplication-behavior:
+
+Duplication behavior:
+~~~~~~~~~~~~~~~~~~~~~
+
+Defines the FAL behavior, when a file with the same name exists in the target
+folder. Possible values are :php:`DuplicationBehavior::RENAME` (default),
```

Comment:
Please use the FQCN of DuplicationBehaviour at least once. I also suggest to use a list for better readability.

Reply:
done